### PR TITLE
Conditionally adds cloudformation template

### PR DESCRIPTION
**Pull request by etgrieco**
Potential solution to issue/request https://github.com/leftclickben/serverless-api-stage/issues/13

### CODE REVIEW

**Comment by leftclickben** [comment]
Thanks for the contributions! So sorry I haven't had a chance to review this yet. I will take a proper look later tonight.

**Comment by leftclickben** [comment]
I agree with #13 and this is a good solution that involves minimal code change.  

However I'm conflicted about a few things:

* The behaviour when you don't specify the `MethodSettings`, whether it should use the "defaults" or not, like you pointed out
* The fact that `DataTraceEnabled` is `true` in the defaults for `MethodSettings` (line 130 on this branch), which itself would imply that the logging permissions are required by default (and probably this shouldn't be the case)
* You have also uncovered another problem with the permissions: it's missing `cloudwatch:PutMetricData`, which is required when `MetricsEnabled` is `true` (and if the permissions are applied conditionally as per this PR, the `logs:*` permissions should be applied only on `LoggingLevel` and `DataTraceEnabled`)
* At it stands, I _think_ this PR is already BC breaking (at least concerning `README.md`, although the existing tests haven't been changed)... I'm not opposed to that, rather I'm thinking we can go further with it, since it's a major version bump already

@etgrieco thanks again for this.  Let me think about the above, and I will get back to you.  I'm leaning towards extending these changes further and making it `2.x`, but it will need a migration guide and support for both major versions, at least for a while.

File: src/index.js [hunk 1]
```diff
@@ -12,93 +12,99 @@ module.exports = function (serverless) {
...
+                        }
+                    },
+                    DependsOn: [
+                        logRoleLogicalName
+                    ]
+                }
+            };
+            const hasCloudFormationSettings = methodSettings.some(setting =>
+                setting === undefined ||
+                ['DataTraceEnabled', 'MetricsEnabled', 'LoggingLevel'].some(prop => setting[prop])
```
Comment (leftclickben) [inline src/index.js:102]: A couple of specific points here:

* `LoggingLevel` can have a value of `OFF`; if the logging permissions are only being applied when logging is enabled, then it seems reasonable that they wouldn't be applied if the value is `OFF` (maybe?)
* As per my overall comment -- and this isn't something you've introduced but rather uncovered -- this should actually be split into two groups, `DataTraceEnabled` and `LoggingLevel` should enable the `logs:*` permissions and `MetricsEnabled` should enable the `cloudwatch:PutMetricData` permission

Comment (etgrieco) [reply src/index.js:102]: @leftclickben I am still a bit green on some of this permissions stuff but what you're saying so far makes sense. Any new ideas recently?

Comment (etgrieco) [reply src/index.js:102]: @leftclickben I also realized that what I want to accomplish is possible with serverless-plugin-stage-variables without needing the extra permissions (I was confused at first about how to implement this) so I am sticking with that as a solution to my specific issue. However, if you do want to be more refined about your permissions on this project then hopefully the input helped!

**Pull request closed without merging**